app/notifications/potential_hire_notification.rb

```diff
@@ -0,0 +1,27 @@
+class PotentialHireNotification < Noticed::Base
+  deliver_by :database, if: :still_needed?
+  deliver_by :email, mailer: "AdminMailer", method: :potential_hire, if: :still_needed?
+
+  param :developer
+
+  def title
+    t "notifications.potential_hire",
+      developer: developer.name
+  end
+
+  def developer
+    params[:developer]
+  end
+
+  def url
+    developer_path(developer)
+  end
+
+  def still_needed?
+    return false if developer.notifications_as_subject
+      .where(created_at: 7.days.ago..10.seconds.ago,
+        type: "PotentialHireNotification").exists?
+
+    !developer.new_developer_account?
+  end
```

Comment:
Can you dive into why this is needed? It feels like a pre-optimization.

Reply:
Oh, and for future reference, I think we can use the built in querying support from noticed to find these.


https://github.com/excid3/noticed/discussions/220#discussioncomment-2501568

Reply:
Addressed in comments below, I agree :)